packages/core/src/node/messaging/messaging-contribution.ts

```diff
@@ -173,9 +173,14 @@ export class MessagingContribution implements BackendApplicationContribution, Me
         });
     }
 
-    protected getConnectionChannelHandlers(socket: ws): MessagingContribution.ConnectionHandlers<WebSocketChannel> {
-        const connectionContainer = this.container.createChild();
+    protected createSocketContainer(socket: ws): Container {
+        const connectionContainer: Container = this.container.createChild() as Container;
```

Comment:
(nit) ts should be able infer types here

Reply:
yes this is what I was thinking as well but without the `as Container` I've

```
src/node/messaging/messaging-contribution.ts(182,15): error TS2740: Type 'Container' is missing the following properties from type 'Container': _middleware, _bindingDictionary, _snapshots, _metadataReader, and 5 more.
```

Reply:
@akosyakov I think it's because there
https://github.com/inversify/InversifyJS/blob/ac59f754fcc4bc29d77804ebb5ae0e8bce90c46d/src/container/container.ts#L220
it's returning `Container` and not `interfaces.Container`

Reply:
ah ok, looks good to merge